core/core.lua

```diff
@@ -297,8 +298,8 @@ local function brainstorm_sprite(brainstorm, card)
 
     local needed_atlas = card.children.floating_sprite and brainstorm_atlas(card.children.center.atlas, card.children.floating_sprite.atlas, offset)
                                                         or brainstorm_atlas(card.children.center.atlas, nil, nil)
-
-    if brainstorm.children.center.atlas.name == needed_atlas.name then
+    
+    if brainstorm.children.center.atlas.name == needed_atlas.name and card.T.x == brainstorm.T.x and card.T.y == brainstorm.T.y then
```

Comment:
this check will update the sprite every frame instead of updating only when changed. You should check for center sprite pos within atlas instead